cmd/bpfman-agent/main.go

```diff
@@ -139,72 +139,95 @@ func main() {
 		os.Exit(1)
 	}
 
-	common := bpfmanagent.ReconcilerCommon{
+	common := bpfmanagent.ReconcilerCommon[bpfmaniov1alpha1.BpfProgram, bpfmaniov1alpha1.BpfProgramList]{
 		Client:       mgr.GetClient(),
 		Scheme:       mgr.GetScheme(),
 		GrpcConn:     conn,
 		BpfmanClient: gobpfman.NewBpfmanClient(conn),
 		NodeName:     nodeName,
 	}
 
-	if err = (&bpfmanagent.XdpProgramReconciler{
+	commonCluster := bpfmanagent.ClusterProgramReconciler{
 		ReconcilerCommon: common,
+	}
+
+	commonNs := bpfmanagent.ReconcilerCommon[bpfmaniov1alpha1.BpfNsProgram, bpfmaniov1alpha1.BpfNsProgramList]{
+		Client:       mgr.GetClient(),
+		Scheme:       mgr.GetScheme(),
+		GrpcConn:     conn,
+		BpfmanClient: gobpfman.NewBpfmanClient(conn),
+		NodeName:     nodeName,
+	}
+
+	commonNamespace := bpfmanagent.NamespaceProgramReconciler{
+		ReconcilerCommon: commonNs,
+	}
+
+	if err = (&bpfmanagent.XdpProgramReconciler{
+		ClusterProgramReconciler: commonCluster,
 	}).SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create xdpProgram controller", "controller", "BpfProgram")
 		os.Exit(1)
 	}
 
+	if err = (&bpfmanagent.XdpNsProgramReconciler{
+		NamespaceProgramReconciler: commonNamespace,
```

Comment:
Do you need to do the same for the other namespace program types?

Reply:
Yes, found that about 10 mins ago. Was missing in the operator too. Fixed in the last push.